x-pack/plugins/fleet/server/saved_objects/index.ts

```diff
@@ -124,6 +124,8 @@ const getSavedObjectTypes = (
       '7.10.0': migrateAgentActionToV7100(encryptedSavedObjects),
     },
   },
+  // TODO: Remove this saved object type. Core will drop any saved objects of
```

Comment:
Because we're dropping any `fleet-agent-events` it would be critical if we started saving something there that was important to keep. As a follow-up to this PR we should remove this saved object type from fleet and once that's done I can add `fleet-agent-events` to a denylist inside core so that future plugins don't accidentally start using this saved object.